worker/dbaccessor/worker.go

```diff
@@ -239,19 +251,37 @@ func (w *dbWorker) loop() (err error) {
 		}
 	}
 
+	timer := w.cfg.Clock.NewTimer(defaultNamespaceCacheFlushInterval)
+	defer timer.Stop()
+
 	for {
 		select {
 		case req := <-w.dbRequests:
+			// Ensure the namespace exists or is allowed to open a new one
+			// before we attempt to open the database.
+			if err := w.ensureNamespace(req.namespace); err != nil {
+				req.done <- errors.Annotatef(err, "ensuring namespace %q", req.namespace)
```

Comment:
I like the potential of the simplicity, but you're just masking errors and returning bogus information. It would be really nice to know why a given namespace fails without having to see the logs.